### PR TITLE
CYB-371 - User interface tweaks

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
- Styling improvements to give the various tools an overall better and more consistent look
- Improved the interactions around sharing your points on the engagement index
- Improved the engagement index instructions

https://jira.ets.berkeley.edu/jira/browse/CYB-371
https://jira.ets.berkeley.edu/jira/browse/CYB-399
https://jira.ets.berkeley.edu/jira/browse/CYB-387

### CODE REVIEW

File: app/assets/stylesheets/_engagement_index.scss [hunk 1]
```diff
@@ -1,10 +1,16 @@
 // BODY
 .dc-engagement-index {
+  table {
```
Comment (christianv) [inline app/assets/stylesheets/_engagement_index.scss]: @nicolaasmatthijs this seems pretty general - would it be possible to use classes instead? (same with the `input`, `button` and `hr`)

File: app/assets/stylesheets/_main.scss [hunk 2]
```diff
@@ -8,11 +8,17 @@ h1, h2, h3 {
...
+  line-height: 40px;
   margin-top: 0;
 }
 h2 {
-  font-size: 24px;
+  font-size: 19px;
+  line-height: 30px;
+}
+h1 a, h2 a {
+  color: inherit;
```
Comment (christianv) [inline app/assets/stylesheets/_main.scss]: @nicolaasmatthijs does this mean that an the link inside and `h1` / `h2` will have the same color as an `h1` / `h2`? If so, isn't that and accessibility issue?

File: app/assets/stylesheets/_points_configuration.scss [hunk 3]
```diff
@@ -4,16 +4,22 @@
...
   .dc-points-configuration-main-container {
-    max-width: 440px;
+    max-width: 550px;
   }
   .dc-points-configuration-table-container {
     margin: 5px auto;
-    table td {
-      vertical-align: middle;
+    table {
+      table-layout: fixed;
```
Comment (christianv) [inline app/assets/stylesheets/_points_configuration.scss]: @nicolaasmatthijs so we'll never need to make this responsive, correct?

File: app/assets/templates/engagement_index_sharing_option.html [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-  <form data-ng-submit="setEngagementIndexStatus()">
+  <form data-ng-submit="saveEngagementIndexStatus()">
     <div>
-      <input type="checkbox" data-ng-model="currStudent.share" data-ng-change="resizeIFrame()" id="dc-engagement-index-choice">
+      <input type="checkbox" data-ng-model="currStudent.share" data-ng-change="changeEngagementIndexStatus()" id="dc-engagement-index-choice">
       <label for="dc-engagement-index-choice">Yes, I want to share my score on the Engagement Index.</label>
     </div>
-    <div>
-      <button type="submit" class="btn btn-info">Save</button>
+    <div data-ng-if="!currStudent.has_answered_share_question">
```
Comment (christianv) [inline app/assets/templates/engagement_index_sharing_option.html]: @nicolaasmatthijs does this mean that the page will completely re-render when you change the `checkbox`?

**Comment by christianv** [comment]
@nicolaasmatthijs one thing I noticed as well in the CSS is that we're overriding more and more elements `input` - `hr` instead of using classes. We don't really have to change anything in this PR but it does make maintenance a bit harder.

File: app/assets/templates/engagement_index_sharing_option.html (hunk 4 above)
Comment (nicolaasmatthijs) [inline app/assets/templates/engagement_index_sharing_option.html]: Only after the initial save. Once that's done, the `Continue` button will no longer be shown and changing the checkbox will immediately update your preference and re-load the engagement index. I think that's a vastly better experience than the current one.

File: app/assets/stylesheets/_main.scss (hunk 2 above)
Comment (nicolaasmatthijs) [inline app/assets/stylesheets/_main.scss]: Fixing CYB-398 makes this redundant

File: app/assets/stylesheets/_points_configuration.scss (hunk 3 above)
Comment (nicolaasmatthijs) [inline app/assets/stylesheets/_points_configuration.scss]: I came up with an alternative solution that no longer requires this

**Comment by nicolaasmatthijs** [comment]
@christianv : Actually, my preference is to use CSS classes as a default as well, unless there's a case to be made for using a more general selector. From looking at the existing CSS, I just thought there was a preference for general selectors. I'll update this PR to use classes where possible.

**Comment by nicolaasmatthijs** [comment]
Finished follow-up. Back to @christianv for next round of review

File: app/assets/javascripts/angular/controllers/galleryController.js [hunk 5]
```diff
@@ -10,7 +10,7 @@
      * Reset the sort and filtering options to the default values. This will
      * also ensure that the page is scrolled to the top when showing the items
      */
-    $scope.resetSortAndFilter = function() {
+    var resetSortAndFilter = function() {
```
Comment (christianv) [inline app/assets/javascripts/angular/controllers/galleryController.js]: @nicolaasmatthijs thanks for making this change as well!

File: app/assets/templates/engagement_index_sharing_option.html (hunk 4 above)
Comment (christianv) [inline app/assets/templates/engagement_index_sharing_option.html]: @nicolaasmatthijs sounds gtm, thanks for verifying.

**Comment by christianv** [comment]
@nicolaasmatthijs Thanks for the follow up, this is good to merge but it looks like we have a merge conflict. Feel free to ping another dev to merge after the conflicts have been resolved.

**Comment by nicolaasmatthijs** [comment]
Followed up on review. This is ready for merging.

**Comment by nicolaasmatthijs** [comment]
@raydavis has volunteered to have a look at this

**Pull request merged**